python/vowpalwabbit/pyvw.py

```diff
@@ -1270,6 +1277,73 @@ def __init__(self, action: int, score: float):
         self.score = score
 
 
+class CCBSlotOutcome:
+    def __init__(
+        self, cost: Union["example", float] = 0, action_probs: List[ActionScore] = []
+    ):
+        if isinstance(cost, example):
+            self.from_example(cost)
+        else:
+            self.cost = cost
+            self.action_probs = action_probs
+
+    def from_example(self, ex: "example"):
+        self.cost = ex.get_ccb_cost()
+        self.action_probs = []
+        for i in range(ex.get_ccb_num_included_actions()):
+            self.action_probs.append(
+                ActionScore(ex.get_ccb_class(i), ex.get_ccb_probability(i))
+            )
+
+    def __str__(self):
+        top_action, top_score = self.action_probs[0].action, self.action_probs[0].score
+        out = "{}:{}:{}".format(top_action, round(self.cost, 2), round(top_score, 2))
+        for action_score in self.action_probs[1:]:
+            out += f",{action_score.action}:{action_score.score}"
+        return out
+
+
+class CCBLabel(AbstractLabel):
+    """Class for conditional contextual bandits VW label"""
+
+    def __init__(
```

Comment:
This will need to be updated the match the new way from this PR: https://github.com/VowpalWabbit/vowpal_wabbit/pull/3615